### PR TITLE
feat(iro): scale the bonding curve

### PR DESCRIPTION
**Pull request by mtsitrin**
## Description

----

Closes #XXX

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

----

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: app/keepers/keepers.go [hunk 1]
```diff
@@ -117,7 +117,6 @@ type AppKeepers struct {
 	ParamsKeeper                  paramskeeper.Keeper
 	IBCKeeper                     *ibckeeper.Keeper // IBC Keeper must be a pointer in the app, so we can SetRouter on it correctly
 	TransferStack                 ibcporttypes.IBCModule
```
Comment (omritoptix) [inline app/keepers/keepers.go:119]: why was this removed?

File: x/iro/types/bonding_curve.go [hunk 2]
```diff
@@ -7,17 +7,49 @@ import (
...
+
+N (exponent) shapes the curve's trajectory. When N > 1, the curve becomes convex, accelerating price growth at higher supply levels, which can create strong incentives for early adoption. When 0 < N < 1, the curve is concave, slowing price growth as supply increases, which can promote more stable long-term growth.
+
+C (constant) sets the starting price when supply is zero, effectively establishing a price floor and influencing the token's initial accessibility.
 */
 
 const (
-	MaxNValue    = 10
-	MaxPrecision = 2
+	MaxNValue     = 2 // Maximum allowed value for the N parameter
```
Comment (omritoptix) [inline x/iro/types/bonding_curve.go:27]: why we changed to 2?

File: x/iro/types/bonding_curve.go [hunk 3]
```diff
@@ -7,17 +7,49 @@ import (
...
+	MaxNPrecision = 3 // Maximum allowed decimal precision for the N parameter
 )
 
+/*
+The bonding curve implementation based on decimal representation of the X (rollapp's tokens) and Y (DYM) values.
+we use scaling functions to convert between the decimal scale and the base denomination.
+*/
+
+// Scales x from it's base denomination to a decimal representation
+// This is used so the bonding curve
```
Comment (omritoptix) [inline x/iro/types/bonding_curve.go:37]: cut-off comment?

**Review by omritoptix — DISMISSED** [review]
non blocker comment and questions

File: app/keepers/keepers.go (hunk 1 above)
Comment (mtsitrin) [reply app/keepers/keepers.go:119]: it's unused. just cleanup

File: x/iro/types/bonding_curve.go (hunk 2 above)
Comment (mtsitrin) [reply x/iro/types/bonding_curve.go:27]: reviewing real world scenarios, can't see how it can be usable
even N=2 is probably too high
the initial N=10 was out of the blue
